honeybadgermpc/protocols/avid.py

```diff
@@ -225,3 +209,17 @@ def broadcast(self, o):
             elif msg[1] == AVIDMessageType.RESPONSE:
                 # put in the queue for retrieve
                 self.retrieval_queue.put_nowait((sender, msg))
+
+            if len(echo_set) >= echo_threshold and not ready_sent:
```

Comment:
Why was this bug not caught via the tests?

Can you add a test so that if this happens again we catch it?

Reply:
I don't think that is necessary. There were some logical misunderstandings while I wrote the code and it's not a bug to catch. And the AVID is fully tested by all different kinds of batch HBAVSS tests anyway.

Reply:
I disagree, unit tests are meant to test things independent of other components. There is not even a single test for AVID. If I modify AVID I don't want to run HbAVSS tests to see if something is broken. You should write tests to test the AVID code.

Reply:
AVID is part of HBAVSS here. It is just put in a separate file.

Reply:
AVID (similar to RBC) as far as I understand is a separate protocol and must have tests. Nonetheless, it is in a separate file implying it is a separate module which can be reused.

If you don't have time, then you can open an issue and add the issue link in this PR so that this can be taken up later. I will approve this PR post that.

Reply:
Sure. Will do.